Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/CosmosServerlessSparkBatchJob.java

```diff
@@ -260,94 +249,143 @@ public String getJobUuid() {
-                                .map(logs -> logs.stream()
-                                        .filter(logLine -> !ignoredEmptyLines.contains(logLine.trim().toLowerCase()))
-                                        .collect(Collectors.toList()))
-                                .flatMap(logLines -> {
-                                    if (logLines.size() > 0) {
-                                        return Observable.just(Pair.of(logLines, SparkBatchJobState.STARTING));
-                                    } else {
-                                        return getResponsePayloadWithState()
-                                                .flatMap(responsePayload ->
-                                                        Observable.just(Pair.of(logLines, responsePayload.getState()))
-                                                        .delay(getDelaySeconds(), TimeUnit.SECONDS));
-                                    }
-                                })
-                                .repeatWhen(ob -> ob.delay(GET_LOG_REPEAT_DELAY_MILLISECONDS, TimeUnit.MILLISECONDS))
-                                // FIXME:
-                                // For HDI spark job, the ending condition is `jobState != starting || appIdIsAllocated`
-                                // However, currently the response has no appId all the time so we ignored the app id check
-                                .takeUntil(logAndStatePair ->
-                                        !logAndStatePair.getRight().toString()
-                                                .equalsIgnoreCase(SparkBatchJobState.STARTING.toString()))
-                                .flatMap(logAndStatePair -> Observable.from(logAndStatePair.getLeft()))
-                                .map(line -> new AbstractMap.SimpleImmutableEntry<>(Log, line))
-                                .retryWhen(error -> error.flatMap(exception -> {
-                                    if (exception instanceof SparkJobFinishedException) {
-                                        throw propagate(exception);
-                                    } else {
-                                        // delay sometime and retry with limited times
-                                        return Observable.just(1)
-                                                .zipWith(Observable.range(1, getRetriesMax()), (n, i) -> i)
-                                                .doOnNext(i -> ctrlInfo("Retry retrieving livy submission log..."))
-                                                .delay(getDelaySeconds(), TimeUnit.SECONDS);
-                                    }
-                                }))
-                )
-                .onErrorResumeNext(err -> {
-                    if (err instanceof SparkJobFinishedException || err.getCause() instanceof SparkJobFinishedException) {
-                        return Observable.error(err);
+                .takeUntil(sparkBatchJob ->
```

Comment:
The val `sparkBatchJob` might be easy confused. You can try `batchResponse` or `batchResp`